### ISSUE TITLE
[P2][algorithm/correctness] RsCrossValidation::kFold fails when classes have < k samples and ignores KMeans cluster remapping

### ISSUE THREAD

**Issue by WindWang2** [WindWang2/exp-rs#413]
## Description

In `src/analysis/classification/rs_cross_validation.cpp`, `RsCrossValidation::kFold` suffers from two critical algorithmic defects:
1. **Total CV collapse on small class sample sizes**: When all classes have fewer samples than the fold count $k$ (e.g. 4 samples per class with 5-fold CV, common in rare-class remote sensing classification), `foldTest` remains completely empty for all $k$ folds. Every fold is skipped as degenerate, and `kFold` fails with `"All folds failed"`.
2. **KMeans Cluster ID vs Class ID mismatch**: For unsupervised clustering (`KMeans`, where `backend->needsLabelRemap() == true`), `kFold` directly compares predicted cluster indices $1..K$ against true ground-truth class IDs (e.g. 10, 20, 30) without Hungarian optimal assignment, unconditionally reporting 0% accuracy.

--------

### 1. Small Sample CV Skip
In `src/analysis/classification/rs_cross_validation.cpp` (lines 59–79 and 104–108):
```cpp
  QVector<QVector<int>> foldTest( k );
  for ( auto it = byClass.constBegin(); it != byClass.constEnd(); ++it )
  {
    const QVector<int> &bucket = it.value();
    if ( bucket.size() < k )
    {
      // Class doesn't have enough samples to spread across folds —
      // keep all in train (foldTest empty for this class).
      continue;
    }
    for ( int i = 0; i < bucket.size(); ++i )
      foldTest[i % k].append( bucket[i] );
  }
```
If all classes have sample count $< k$, `foldTest[fi]` is empty for all $fi \in [0, k-1]$.
Then at line 104:
```cpp
  if ( trainIdx.isEmpty() || testIdx.isEmpty() )
  {
    // Degenerate fold; skip.
    continue;
  }
```
Every fold is skipped, `accs` is empty, and `kFold` returns an error.

### 2. KMeans Cluster Evaluation
At lines 158–162:
```cpp
  int correct = 0;
  for ( int i = 0; i < pred.rows; ++i )
    if ( pred.at<int>( i, 0 ) == testY.at<int>( i, 0 ) )
      ++correct;
  accs.append( double( correct ) / pred.rows );
```
When `backend->needsLabelRemap()` is true, `pred` outputs cluster labels $1..k$, but `testY` contains ground-truth class IDs (e.g., 10, 20). No Hungarian assignment is performed to map cluster IDs to ground-truth IDs prior to accuracy calculation.

--------

## Proposed Fix

1. For small-sample classes ($< k$), distribute the available samples evenly into the first $N$ folds (`i % k`) rather than completely omitting them from `foldTest`.
2. When `backend->needsLabelRemap()` is true, build the cost matrix on the fold's training set, solve Hungarian assignment via `RsHungarianAssignment::solve`, and remap `pred` labels before evaluating fold accuracy.

**Comment by WindWang2** [comment]
Fixed and verified across test suites in local branch.
